cadquery/cq.py

```diff
@@ -3441,48 +3480,130 @@ def loft(
-        eDir = self.plane.zDir.multiply(distance)
+        if upToFace is not None:
+            eDir = self.plane.zDir
+        elif distance is not None:
+            eDir = self.plane.zDir.multiply(distance)
+
+        if additive:
+            direction = "AlongAxis"
+        else:
+            direction = "Opposite"
 
         # one would think that fusing faces into a compound and then extruding would work,
-        # but it doesn't-- the resulting compound appears to look right, ( right number of faces, etc)
+        # but it doesnt-- the resulting compound appears to look right, ( right number of faces, etc)
         # but then cutting it from the main solid fails with BRep_NotDone.
         # the work around is to extrude each and then join the resulting solids, which seems to work
 
         # underlying cad kernel can only handle simple bosses-- we'll aggregate them if there are
         # multiple sets
+        thisObj: Union[Solid, Compound]
 
         toFuse = []
+        taper = 0.0 if taper is None else taper
+        baseSolid = None
 
-        if taper:
-            for ws in wireSets:
-                thisObj = Solid.extrudeLinear(ws[0], [], eDir, taper)
-                toFuse.append(thisObj)
-        else:
-            for ws in wireSets:
-                thisObj = Solid.extrudeLinear(ws[0], ws[1:], eDir)
+        for ws in wireSets:
+            if upToFace is not None:
+                baseSolid = self.findSolid() if baseSolid is None else thisObj
+                if isinstance(upToFace, int):
+                    facesList = getFacesList(eDir, additive, both=both)
+                    limitFace = facesList[upToFace]
+                    if (
+                        baseSolid.isInside(ws[0].Center())
+                        and additive
+                        and upToFace == 0
+                    ):
+                        upToFace = 1  # extrude until next face outside the solid
+                else:
+                    limitFace = upToFace
+
+                thisObj = Solid.dprism(
+                    baseSolid,
+                    Face.makeFromWires(ws[0]),
+                    ws,
+                    taper=taper,
+                    upToFace=limitFace,
+                    additive=additive,
+                )
+
+                if both:
+                    facesList2 = getFacesList(eDir.multiply(-1.0), additive, both=both)
+                    limitFace2 = facesList2[upToFace]
+                    thisObj2 = Solid.dprism(
+                        self.findSolid(),
+                        Face.makeFromWires(ws[0]),
+                        ws,
+                        taper=taper,
+                        upToFace=limitFace2,
+                        additive=additive,
+                    )
+                    thisObj = Compound.makeCompound([thisObj, thisObj2])
+                toFuse = [thisObj]
+            else:
+                thisObj = Solid.extrudeLinear(ws[0], ws[1:], eDir, taper=taper)
```

Comment:
This line changes the previous behavior - is it intended?

Reply:
Are you referring to the `ws[1:]` being passed in for a tapered extrude ?
I didn't noticed there were this difference in the old behaviour, I will modify that.

Reply:
Yes, that is what I mean. Did you resolve it in the newest commits?

Reply:
Yes it's now handled like it is on master !